### ISSUE TITLE
Footnote references before punctuation

### ISSUE THREAD

**Issue by Jmuccigr** [jgm/pandoc-citeproc#256]
I just had to submit something to a journal that requires footnote references before the final period of a sentence. With regular footnotes (`[^footnote_ref]`), I noticed that if there's a space between the last word of the sentence and the reference, like `lastword [^ref].` **and** I'm using my bibliography file via `--bibliography=filename.json`, the footnote reference is placed after the period, not before it. The same thing happens with inline footnotes.

If there is no space or if I don't use a bibliography, the reference comes before the period and, in the case where there's a space, that space is present. I suspect therefore that it's pandoc-citeproc and not just pandoc.

It seems to me that the behavior shouldn't change when I use a bibliography, and that before-the-period footnotes should behave like after-the-period ones (i.e., there shouldn't be a space before the reference, even when one is put in).

Here's an example to which I applied `pandoc` or `pandoc --bibliography=filename.json`:

```
Here's a footnote before the period and immediately after the last word[^f].

[^f]: Text of the footnote.

Here's one with a preceding space [^f2].

[^f2]: More footnote text.

Here's one with an embedded footnote and no space^[Footnote 3.].

Here's an embedded one with a space ^[Footnote 4.].
```

Another thing involves commas and I'm not sure what the desired behavior is, but when there's a citation that automatically gets a footnote, that reference comes after a comma and I don't think there's a way to change that (again, if it is desirable). So something like `@author1999, p. 999` becomes "Author,<sup>ref</sup>" instead of "Author<sup>ref</sup>,".

**Comment by jgm** [comment]
The intent behind this behavior is to allow the same source
to be used, without modification, with a footnote style
and an author-date style.  So you write

    blah blah [@cite, p. 13].

and it will look right in both note and author-date styles.

Of course this requires moving the citation til after the
period, in almost every style, so pandoc-citeproc does that.

If we didn't do it, manual modification would be needed to
the source to change to a footnote style.

I suppose a better behavior would be to make this sensitive,
somehow, to something in the CSL style.  But I don't think
anything in the style tells you whether note references go
before or after a period.

**Comment by Jmuccigr** [comment]
I think you're right that the before/after issue is not in csl. For that reason, I suspect either it would need to specified elsewhere or via punctuation/layout.

But I'm confused about your example. Using pandoc without specifying any csl results in an author-date style where the citation comes before the period for me (without a footnote, of course), with the space or its absence being preserved. Whereas I get the same thing as before when I specify a footnote csl.

Also, my examples don't have any citations, yet they get affected. (My closing example with the comma is another story and best left alone for now.)

**Comment by jgm** [comment]
+++ John Muccigrosso [Sep 15 16 13:53 ]:
>   I think you're right that the before/after issue is not in csl. For
>   that reason, I suspect either it would need to specified elsewhere or
>   via punctuation/layout.
>
>   But I'm confused about your example. Using pandoc without specifying
>   any csl results in an author-date style where the citation comes before
>   the period for me (without a footnote, of course), with the space or
>   its absence being preserved. Whereas I get the same thing as before
>   when I specify a footnote csl.

Yes.  And if you use a footnote CSL with the same source,
you'll get the citation moved after the period, and the
space that would otherwise come before the period removed.
That's a feature (except for the rather unusual styles
that want the note before the punctuation).

>   Also, my examples don't have any citations, yet they get affected. (My
>   closing example with the comma is another story and best left alone for
>   now.)

Ah, yes.  pandoc-citeproc does this transformation (moving
footnotes) only after citations have been resolved into
footnotes.  So it will affect all footnotes, not just those
produced by citations.  I could look into whether this could
be changed.

**Comment by Jmuccigr** [comment]
Thanks for pursuing this, John.

I think my comments below pertain only to footnote styles, though I'm concerned about footnotes in general, not just those from citations.

What still confuses me is that pandoc-citeproc is not moving references when the notes are not preceded by a space and come before a period. See my example below.

I do think there should be some way of indicating whether references should go before or after the sentence-final period, and simply putting the note before or after the period seems like a reasonable way to do it. But having the behavior change based on the presence of a preceding space seems undesirable. To my mind, when the note precedes the period, any preceding space should be removed, but the reference should stay before the period. This mimics (I think) the behavior when the note come after the period.

An example:
```{markdown}
This sentence has a footnote without a space before it^[This is a footnote.].
This sentence has a footnote with a space before it ^[This is also a footnote.].
```

results in this (with a couple of returns added for readability):
```{html}
<p>
This sentence has a footnote without a space before it<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a>. 
This sentence has a footnote with a space before it.<a href="#fn2" class="footnoteRef" id="fnref2"><sup>2</sup></a>
</p>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>This is a footnote.<a href="#fnref1">↩</a></p></li>
<li id="fn2"><p>This is also a footnote.<a href="#fnref2">↩</a></p></li>
</ol>
</div>
```

**Comment by jgm** [comment]
The program is designed with the thought that, in the
source, you'll have references that look like author-date
references -- and here you'd always have a space
before it.

As I said, I didn't intend to move footnotes that were
explicitly placed (rather than generated from references)
would be moved at all.

Will have to look at this.

+++ John Muccigrosso [Sep 16 16 05:03 ]:

**Comment by zorrillodepeluche** [comment]
The position of the footnote-call relative to the punctuation doesn't depend on the bibliography style, but on the language in which one is writing. See, for example the [United Nations Editorial Manual](http://dd.dgacm.org/editorialmanual/ed-guidelines/footnotes/footnotes_chap_09.htm#B), that indicate that, for French and Spanish (my native language) the footnote-call must be placed before the punctuation (while, for English, it should be after).

I would like to request a way (maybe in the metadata) to set the following behaviour: That, if using a footnote style, `pandoc-citeproc` remove the space, then place the footnote, then the punctuation.

This may even be a default if French is set as `lang`. In spanish is prevalent, but not mandatory, and usually depends on personal style or the journal's guidelines.

**Comment by Jmuccigr** [comment]
Well, it might depend on the traditions of the country of the journal's origins. My example was from an English-language article in a journal based in Italy.

In any case, I think some kind of option, placeable in either the YAML or command line, would be helpful.

**Comment by jgm** [comment]
One possibility would be to add a metadata field
`notes-after-punctuation` (boolean defaulting to True).

How does that sound?

We would still collapse the space before the note,
I assume.

+++ zorrillodepeluche [Oct 04 16 14:20 ]:

**Comment by zorrillodepeluche** [comment]
>One possibility would be to add a metadata field `notes-after-punctuation` (boolean defaulting to True).

That sounds very well, and yes, I think the space should still be collapsed to remain compatible with author-date styles.

Thank you.
